### PR TITLE
Fix normalizeURL to not beak URLs containing paths

### PR DESCRIPTION
**Pull request by jayme-github**
normalizeURL used to break URLs with path by rewriting teir host part.
I'm not sure in which situations that code path is needed, but tests run fine without it.

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -341,9 +341,6 @@ func normalizeURL(addr string) (*url.URL, error) {
 	if u.Opaque != "" {
 		u.Host = net.JoinHostPort(u.Scheme, u.Opaque)
 		u.Opaque = ""
-	} else if u.Path != "" && !strings.Contains(u.Path, ":") {
-		u.Host = net.JoinHostPort(u.Path, "8888")
-		u.Path = ""
 	} else if u.Scheme == "" {
```
Comment (lavagetto) [inline api/client/client.go]: While we're at it, should we fix the indentation of this if stanza and the ones below?

File: api/client/client_test.go [hunk 2]
```diff
@@ -50,6 +51,18 @@ func TestNewServer(t *testing.T) {
...
+	hosts = s.Hosts()
+	if len(hosts) != 1 || hosts[0] != "https://1.1.1.1:8181/" {
+		t.Fatalf("expected [https://1.1.1.1:8181/], but have %v", hosts)
+	}
+
+	s = NewServer("https://1.1.1.1/foo")
+	hosts = s.Hosts()
+	if len(hosts) != 1 || hosts[0] != "https://1.1.1.1/foo" {
+		t.Fatalf("expected [https://1.1.1.1/foo], but have %v", hosts)
+	}
```
Comment (lavagetto) [inline api/client/client_test.go:52]: Please add a test for '1.1.1.1:888', '1.1.1.1/foo'.

**Review by lavagetto — CHANGES_REQUESTED** [review]
Please add additional test cases as outlined in the comments.

File: api/client/client_test.go (hunk 2 above)
Comment (jayme-github) [reply api/client/client_test.go:52]: This revealed more shortcomings of normalizeURL who's functionality is now mostly replaced by urlx.Parse()

**Review by lavagetto — APPROVED** [review]
LGTM!

**Pull request merged**